src/ui/hooks/useSetNodesDataMutation.ts

```
@@ -4,24 +4,47 @@ import {
+        // Patch the page-wide cache directly so returning to Index does not
+        // trigger a full-page rescan. useConnectedNodes uses staleTime:30s,
+        // so fresh data here prevents the redundant allNodes.refetch() call.
+        const key = [getConnectedNodesEndpoint.name, true] as const;
+        const old = queryClient.getQueryData<CachedNodes>(key);
+        if (old) {
+          const patch = new Map(nodes.map((n) => [n.id, n]));
+          const updated = old.items
+            .map((item) =>
+              patch.has(item.id)
+                ? { ...item, ...patch.get(item.id)! }
+                : item,
```

ESLint

[error] Replace `⏎················?·{·...item,·...patch.get(item.id)!·}⏎···············` with `·?·{·...item,·...patch.get(item.id)!·}`

(prettier/prettier)